design/client.md

```diff
@@ -0,0 +1,23 @@
+# Client
+
+Since containerd uses [gRPC](https://grpc.io/), it should be easy to implement 3rd-party containerd client in almost any language.
+
+<!--- TODO: more information about how containerd is a client-heavy design --->
+
+However, 3rd-party client would need to reimplement some operations, which are not implemented on the daemon side:
+
+- Pull & Push images from remote registries, such as OCI images via Docker Registry
+- Import & Export local image archives
+- Generate container runtime configuration, such as OCI Runtime Spec
+- Chown files under the container rootfs for remapping UID / GID
+
+## Label conventions
+
+Client implementations SHOULD follow this labeling convention so as to promote compatibility across different implementations.
+
+Key (omitted prefix: `containerd.io/`)  | Object                                           | Value
+----------------------------------------|--------------------------------------------------|--------------------------------------------------
+`uncompressed`                          | Content (compressed layer blobs), Snapshot       | Digest string of uncompressed layer blob.
+`gc.root`                               | Content, Snapshot                                | RFC3339 UTC timestamp string. Objects with this label are treated as the roots for GC.
```

Comment:
These aren't restricted to these types, are they?

Reply:
@dmcgowan PTAL (why do we need timestamp here?)

Reply:
@AkihiroSuda The timestamp tells you when the label expires, letting you reserve objects before they are referenced elsewhere.

Reply:
Right now I am using the creation time. While garbage collection is done by the serve, a "prune" operation will be client side and these timestamps can be used to hint any client doing pruning as to which objects may be old. It is a client decision but by adding a timestamp here it gives clients more information it may need. This value is not read by the server, that should be noted, it is only for client use.